### ISSUE TITLE
Problems taking too long to load

### ISSUE THREAD

**Issue by k3n3nna** [davidzajac1/zillacode#8]
I'm able to access Zillacode but when I try and open a problem, the problem remains in the process of loading but never actually loads up.

**Comment by AlanDataPortfolio** [comment]
Im running on Macos, there are a few processes running in port 5000. Even if I kill them, they will reopen under a different PID in the same port 5000.

**Comment by davidzajac1** [comment]
> Im running on Macos, there are a few processes running in port 5000. Even if I kill them, they will reopen under a different PID in the same port 5000.

Zillacode needs port 5000 (and a few others to run) so you have to make sure those ports are open or it won't be able to run.

This should work to kill anything running on the ports it needs on Mac. Please let me know if this works:

```bash
kill -9 $(lsof -ti:5000,5173,8001,8002,8003)
```

**Comment by AlanDataPortfolio** [comment]
I understand that Zillacode needs port 5000 to work. And I did kill the process running in port 5000, however, the process running in port 5000 seems to be a system managed one, meaning it will reopen itself in port 5000 again after I kill it. Is there a workaround?

**Comment by davidzajac1** [comment]
@AlanDataPortfolio I just made a new branch called `uncommon-ports` which has the fix that I think should work for you. Can you try cloning that branch then run `make up` please (might take a few minutes to build all the images)?

```bash
git clone -b uncommon-ports https://github.com/davidzajac1/zillacode.git
make up
```

**Comment by AlanDataPortfolio** [comment]
Thanks David! the `uncommon-ports` branch has fixed the issue on Macos.
Cheers!

**Comment by k3n3nna** [comment]
Stll having issues opening up the problems.  I'm running the makefile from the new branch.  I've checked the network tab and I'm getting the folllowing error:  Access to XMLHttpRequest at 'http://localhost:5000/get_problem' from origin 'http://localhost:5173' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

**Comment by davidzajac1** [comment]
> Stll having issues opening up the problems. I'm running the makefile from the new branch. I've checked the network tab and I'm getting the folllowing error: Access to XMLHttpRequest at 'http://localhost:5000/get_problem' from origin 'http://localhost:5173' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

@k3n3nna if you open Docker Desktop when this is happening are all of the containers green and seem to be running properly?

**Comment by vickytechkey** [comment]
<img width="556" alt="image" src="https://github.com/user-attachments/assets/654824d8-9c26-413c-bbb4-ef66c503cbc6">

**Comment by vickytechkey** [comment]
im also facing same issue in mac though i tried uncommon branch

**Comment by abdelhaqs** [comment]
I'm facing the same issue, I'm running the app in Github code spaces, the questions take too much time to load : 

![image](https://github.com/user-attachments/assets/11784f6b-e094-4b4a-b644-8ad52fe1af66)

![image](https://github.com/user-attachments/assets/1d8bfe10-8b44-4f52-b2e9-89f46193c5ba)

**Comment by The-Powder-of-Data** [comment]
> Thanks David! the `uncommon-ports` branch has fixed the issue on Macos. Cheers!

this also resolved for me, thanks

**Comment by davidzajac1** [comment]
@k3n3nna @vickytechkey @abdelhaqs It looks like you are still having issues because requests from the browser are still being sent to port 5000 instead of port 5001. I can see this in some of the screenshots y'all have posted. 

This is probably because you haven't rebuilt the zillacode-frontend Docker Image locally based off that branch I made with the fixes to change everything from port 5000 to 5001. I have now pushed the changes to Docker Hub and updated the `master` branch so all branches have the fix.

To fix this please completely re-clone the Zillacode `master` branch, then open Docker Desktop -> Images and delete any zillacode-frontend and zillacode-backend images. Then run `docker-compose up` again. When you inspect the page in Chrome you must make sure NO requests are being sent to http://localhost:5000.

**Comment by slowLatency** [comment]
Hi, 
I've cloned the latest master branch, but i'm still facing the same issue. Can someone plz help me?

**Comment by davidzajac1** [comment]
> Hi, I've cloned the latest master branch, but i'm still facing the same issue. Can someone plz help me?

@slowLatency yes looks like there's actually still an issue. I just pushed a fix to the `master` branch and to DockerHub.

Can you rebase the `master` branch you have cloned, delete any zillacode-backend Docker images you have then try running `docker-compose up` again please?

**Comment by slowLatency** [comment]
@davidzajac1 I cloned the master branch once again, deleted all the docker container and re-ran the docker-compose up, i'm still facing the same issue :(

**Comment by davidzajac1** [comment]
> @davidzajac1 I cloned the master branch once again, deleted all the docker container and re-ran the docker-compose up, i'm still facing the same issue :(

@slowLatency can you try one more time please? I just changed something in Docker Hub so I think it will work now.

**Comment by slowLatency** [comment]
Hi @davidzajac1, it won't work T_T

**Comment by davidzajac1** [comment]
> Hi @davidzajac1, it won't work T_T

@slowLatency did you delete all of the zillacode Images in Docker Desktop? Are requests still being sent to port 5000 when you Inspect the page? Thanks for being patient with this!

I'm guessing somehow you don't have the newest Docker Images pulled locally. Your Docker desktop should look like this with no zillacode related images in it before trying to run `docker compose up` again.

![image](https://github.com/user-attachments/assets/ac92bc01-32e5-4bff-88e4-22d851287d9a)

**Comment by slowLatency** [comment]
Hi @davidzajac1 it worked. I had deleted the containers but i did not know that even after you delete the containers, images are still present. Rookie move on my part( I'm learning docker c: ).
Thank you for your help
